components/forms/Multiselect/Multiselect.styles.ts

```diff
@@ -47,11 +49,18 @@ export const List = styled.ul<ListProps>`
   max-height: 50vh;
   overflow-y: auto;
   transform: translateY(${pxToRem("41px")});
-  z-index: var(--z-index-dropdown);
 
-  ${noInsetSupport(`left: 0; right: 0; top: 0;`)}
+  display: none;
+  pointer-events: none;
+  z-index: -1;
 
-  ${({ isOpen }) => !isOpen && `opacity: 0;`}
```

Comment:
This was the culprit - passing in a property to the styled component cause the component to re-render unexpectedly. Once I switched to checking for attributes (`&[open]`), the issue was fixed.